assets/evaluation_on_cloud/environments/evaluations-built-in/context/model_target.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Module for handling model targets in evaluation on cloud."""
+import requests
+import logging
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+class ModelTarget:
+    """Represents a model target and runs the chat/completion API on a query based on the target."""
+
+    def __init__(self, endpoint, api_key, model_params, system_message, few_shot_examples):
+        """Initialize ModelTarget."""
+        self.endpoint = endpoint
+        self.api_key = api_key
+        self.model_params = model_params
+        self.system_message = system_message
+        self.few_shot_examples = few_shot_examples
+
+    def __call__(self, query, **kwargs):
+        """Invoke the model target with the given input query."""
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": "Bearer " + self.api_key
+        }
+
+        messages = [{"role": "system", "content": self.system_message}]
+        for example in self.few_shot_examples:
+            if 'User' in example:
+                messages.append({"role": "user", "content": example['User']})
+            if 'Assistant' in example:
+                messages.append({"role": "assistant", "content": example['Assistant']})
+        messages.append({"role": "user", "content": query})
+
+        payload = {
+            "messages": messages,
+            **self.model_params
+        }
+
+        response = requests.post(self.endpoint, headers=headers, json=payload)
+
+        if response.status_code == 200:
+            try:
+                output = response.json()
+                response_text = output["choices"][0]["message"]["content"]
+                return {"query": query, "response": response_text}

```

Comment:
This I believe should have context and ground_truth

Reply:
Synced with Qun - 
We think there are 2 issues: 
1. ground_truth and context passed by client is getting dropped here
2. We dont have the final data_mapping stored now, so before you make the call to "run_evaluation", log the data_mapping

Reply:
moved response_feneration here inside env and passing new data_file to evaluate()